pipeline/relation_embedder/batcher/src/main/scala/weco/pipeline/batcher/LambdaMain.scala

```diff
@@ -2,18 +2,52 @@ package weco.pipeline.batcher
 
 import com.amazonaws.services.lambda.runtime.{Context, RequestHandler}
 import grizzled.slf4j.Logging
+import com.amazonaws.services.lambda.runtime.events.SQSEvent
+import com.amazonaws.services.lambda.runtime.events.SQSEvent.SQSMessage
+import org.apache.pekko.actor.ActorSystem
+import weco.messaging.typesafe.SNSBuilder
+import weco.json.JsonUtil._
+import com.typesafe.config.ConfigFactory
+import weco.typesafe.config.builders.EnrichConfig.RichConfig
 
-import java.util.{Map => JavaMap}
+import scala.collection.JavaConverters._
+import scala.concurrent.ExecutionContext
+import scala.util.Try
 
-object LambdaMain
-    extends RequestHandler[JavaMap[String, String], String]
-    with Logging {
+object LambdaMain extends RequestHandler[SQSEvent, String] with Logging {
+  private val config = ConfigFactory.load("application")
 
   override def handleRequest(
-    event: JavaMap[String, String],
+    event: SQSEvent,
     context: Context
   ): String = {
     info(s"running batcher lambda, got event: $event")
+
+    implicit val actorSystem: ActorSystem =
+      ActorSystem("main-actor-system")
+    implicit val ec: ExecutionContext =
+      actorSystem.dispatcher
+
+    val recordList: List[SQSMessage] = event.getRecords.asScala.toList
+    val paths = recordList map {
+      message: SQSMessage =>
+        // This assumes that the input is message body is just the path
```

Comment:
My understanding is that an SQS event comes in this [format](https://docs.aws.amazon.com/lambda/latest/dg/with-sqs.html#example-standard-queue-message-event)
`event.getRecords` should give us the list of messages
Then for each message, `message.getBody` gives us "The message's contents (not URL-encoded)" 
The event's messages' bodies is what we want here isn't it?

Reply:
I've subscribed to both topics that feed into the batch_input queue, let's see what comes in

Reply:
Sorted.  This has also necessitated the script mentioned [here](https://github.com/wellcomecollection/catalogue-pipeline/pull/2735#discussion_r1815188857), because the input is now far too verbose and awkward.